### ISSUE TITLE
Prophecy exception hidden by generic exception catch in application code 

### ISSUE THREAD

**Issue by fabianaromagnoli** [phpspec/prophecy#301]
Hi,
given this method of a class under test:

```
public function run(array $items)
{
    foreach ($items as $item) {
        try {
            $this->doSomethingThatCouldThrowAnyException($item);
            $this->myProphecy->doSomething();
        } catch (Exception $e) {
            continue;
        }
    }
}
```

I have a test that mocks myProphecy behaviour. If I don't set any expectation on this mock, prophecy throws a \Prophecy\Exception\Call\UnexpectedCallException, but this exception is caught and hidden by the catch in the application code.
I need to catch a generic exception, and not a specific one, because I need the iteration to process all items, even if something for a specific item goes wrong.
What can I do to make test properly fail when expectations are not met?

**Comment by stof** [comment]
Well, if your code swallows all exceptions silently, there is nothing you can do, except stopping this. Btw, the same will be true for all other mocking libraries out there AFAIK.

**Comment by fabianaromagnoli** [comment]
Ok, I suspected that there is nothing to do, but I wanted to be 100% sure.
Thanks a lot for your answer.

**Comment by ciaranmcnulty** [comment]
You could rethrow prophecy exceptions:

```
        } catch (Exception $e) {
            if ($e instanceof \Prophecy\Exception) {
                throw $e;
            }
            continue;
        }
```

**Comment by fabianaromagnoli** [comment]
Yes, this is a possible solution, but we discarded it because we don't want to mess testing code in application code.
Thanks for the suggestion.
